From 3d2f29c88f9b073cb0fd3b9c7f85430e2170acbb Mon Sep 17 00:00:00 2001
From: Masakazu Kitajo <maskit@apache.org>
Date: Tue, 1 Apr 2025 12:15:16 -0600
Subject: [PATCH] Require the use of CRLF in chunked message body (#12150)

* Require the use of CRLF in chunked message body

* Fix docs
---
 doc/admin-guide/files/records.config.en.rst   |  9 +++
 .../functions/TSHttpOverridableConfig.en.rst  |  1 +
 .../api/types/TSOverridableConfigKey.en.rst   |  1 +
 include/ts/apidefs.h.in                       |  1 +
 mgmt/RecordsConfig.cc                         |  2 +
 plugins/lua/ts_lua_http_config.c              |  2 +
 proxy/http/HttpConfig.cc                      |  2 +
 proxy/http/HttpConfig.h                       |  1 +
 proxy/http/HttpSM.cc                          | 28 +++++---
 proxy/http/HttpTunnel.cc                      | 72 ++++++++++++-------
 proxy/http/HttpTunnel.h                       | 15 +++-
 src/shared/overridable_txn_vars.cc            |  1 +
 src/traffic_server/FetchSM.cc                 |  3 +-
 src/traffic_server/InkAPI.cc                  |  3 +
 src/traffic_server/InkAPITest.cc              |  3 +-
 .../malformed_chunked_header.replay.yaml      | 44 ++++++++++++
 16 files changed, 150 insertions(+), 38 deletions(-)

diff --git a/include/ts/apidefs.h.in b/include/ts/apidefs.h.in
index 1641565a1a9..893177c88b9 100644
--- a/include/ts/apidefs.h.in
+++ b/include/ts/apidefs.h.in
@@ -875,6 +875,7 @@ typedef enum {
   TS_CONFIG_HTTP_ENABLE_PARENT_TIMEOUT_MARKDOWNS,
   TS_CONFIG_HTTP_DISABLE_PARENT_MARKDOWNS,
   TS_CONFIG_HTTP_DROP_CHUNKED_TRAILERS,
+  TS_CONFIG_HTTP_STRICT_CHUNK_PARSING,
   TS_CONFIG_LAST_ENTRY
 } TSOverridableConfigKey;
 
